lightning/src/chain/keysinterface.rs

```diff
@@ -195,18 +195,6 @@ impl Readable for SpendableOutputDescriptor {
 // TODO: We should remove Clone by instead requesting a new ChannelKeys copy when we create
 // ChannelMonitors instead of expecting to clone the one out of the Channel into the monitors.
 pub trait ChannelKeys : Send+Clone {
-	/// Gets the private key for the anchor tx
-	fn funding_key<'a>(&'a self) -> &'a SecretKey;
-	/// Gets the local secret key for blinded revocation pubkey
-	fn revocation_base_key<'a>(&'a self) -> &'a SecretKey;
-	/// Gets the local secret key used in the to_remote output of remote commitment tx (ie the
-	/// output to us in transactions our counterparty broadcasts).
-	/// Also as part of obscured commitment number.
-	fn payment_key<'a>(&'a self) -> &'a SecretKey;
-	/// Gets the local secret key used in HTLC-Success/HTLC-Timeout txn and to_local output
-	fn delayed_payment_base_key<'a>(&'a self) -> &'a SecretKey;
-	/// Gets the local htlc secret key used in commitment tx htlc outputs
-	fn htlc_base_key<'a>(&'a self) -> &'a SecretKey;
 	/// Gets the commitment seed
 	fn commitment_seed<'a>(&'a self) -> &'a [u8; 32];
```

Comment:
Future work, should we move `commitment_seed` out-of-memory and behind the interface by moving inside `chan_utils::build_commitment_secret` ? In case of undetected node compromise, access to the commitment seed let you derive revocation secret for future non-yet-existent local commitment transactions ?

Reply:
Yea, I think we have to. If you want an external signer to prevent the host from burning all the coins, you need some kind of exchange here, but we'd also need to get the ordering right, like, to call get_commitment_revocation_secret we'd have to first provide a copy of the new commitment transaction.